tez-dag/src/main/java/org/apache/tez/dag/history/events/AMLaunchedEvent.java

```diff
@@ -90,7 +91,7 @@ public void toProtoStream(CodedOutputStream outputStream) throws IOException {
 
   @Override
   public void fromProtoStream(CodedInputStream inputStream) throws IOException {
-    AMLaunchedProto proto = inputStream.readMessage(AMLaunchedProto.PARSER, null);
+    AMLaunchedProto proto = inputStream.readMessage(AMLaunchedProto.parser(), ExtensionRegistry.newInstance());
```

Comment:
1. is this PARSER -> parser() change backward compatible?
2. what is ExtensionRegistry.newInstance() about exactly?

in general, I want to avoid every direct change in the first place which makes tez unable to compile on 2.5...

Reply:
1. The PARSER field is deprecated in version 3 but replacing it with parser() is breaking backward compatibility so this needs to be reverted.
2. Protobuf3 introduced a nullcheck on the extensionRegistry parameter so if it gets null value then it will throw a `NullpointerException`, that’s why I replaced it with `ExtensionRegistry.newInstance()` which is creating an empty instance (after checking this `getEmptyRegistry()` would be better since it is creating empty maps for the new instance)
I checked the generated code both on Protobuf2 and Protobuf3 but it seems that `extensionRegistry` param is not used in our inputStream.readMessage use case.

Reply:
thanks @mbathori-cloudera, I think now we can move on with PARSER (even if it's deprecated), as we haven't decided about a major tez release yet, so backward compatibility now looks more important to me